stactools_sentinel1/stactools/sentinel1/rtc_metadata.py

```diff
@@ -0,0 +1,166 @@
+from typing import List, Optional
+
+import pystac
+from pystac.utils import str_to_datetime
+
+import rasterio
+import geopandas as gpd
+import os
+import io
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class RTCMetadata:
+    def __init__(self, href):
+        self.href = href
+
+        def _load_metadata(asset='local_incident_angle.tif'):
+            ''' key metadata stored in Geotiff tags '''
+            with rasterio.Env(AWS_NO_SIGN_REQUEST='YES',
+                              GDAL_DISABLE_READDIR_ON_OPEN='EMPTY_DIR'):
+                with rasterio.open(os.path.join(href, asset)) as src:
+                    metadata = src.profile
+                    metadata.update(src.tags())
+                    # other things that aren't keys in src.profile
+                    metadata['PROJ_BBOX'] = list(src.bounds)
+                    metadata['SHAPE'] = src.shape
+
+            return metadata
+
+        def _get_geometries():
+            ''' bbox is MGRS grid square, but geometry is intersection of bbox
+            and S1 frames use to create RTC product. EPSG:4326'''
```

Comment:
Just a note that this overestimates the size of the valid data. Here's an example with one scene.

![image](https://user-images.githubusercontent.com/15164633/112874935-c88f9a00-9080-11eb-8b7a-e6f9d7771752.png)

The purple is the original S1 swath, the orange is the tile, and the brown is the valid data. So if you used this overlay method to create STAC items, and used those to create a mosaic, you might have seams with empty data. Though maybe that seam is small enough.

Reply:
right, thanks for looking into this @kylebarron . the complication here is that the GRD used for the RTC product have lateral nodata borders which the footprint frames encompass, so this is definitely an 'overestimated footprint' approach. the seems can be quite big (~5km) see below screenshot from QGIS. I suspect the footprint should be as accurate and simple as possible?... without doing too much processing (https://rasterio.readthedocs.io/en/latest/topics/features.html#extracting-shapes-of-raster-features).